virtual_ecosystem/core/core_components.py

```diff
@@ -534,6 +511,61 @@ def from_template(self, array_name: str | None = None) -> DataArray:
 
         return template_copy
 
+    @property
+    def index_above(self):
```

Comment:
Don't they need type annotation for the output?

Reply:
Huh - nothing locally complained.. Good point!